torch/utils/_pytree.py

```diff
@@ -180,11 +180,10 @@ class _SerializeNodeDef(NamedTuple):
 
     _cxx_pytree_dynamo_traceable = Version(_optree_version) >= Version("0.13.0")
     if not _cxx_pytree_dynamo_traceable:
-        warnings.warn(
+        raise RuntimeError(
             "optree is installed but the version is too old to support PyTorch Dynamo in C++ pytree. "
             "C++ pytree support is disabled. "
-            "Please consider upgrading optree using `python3 -m pip install --upgrade 'optree>=0.13.0'`.",
-            FutureWarning,
+            "Please upgrade optree using `python3 -m pip install --upgrade 'optree>=0.13.0'`."
         )
```

Comment:
I think we should raise `ImportError` in `torch/utils/_cxx_pytree.py` or revert PR #110670.

- #110670

Reply:
@XuehaiPan are you OK with raising importerror in `torch/utils/_cxx_pytree.py` if optree < 0.13.0 ? I'm not sure if this is actually the true minimum version or if there is nuance